### PR TITLE
Upgrade Python 3.11

### PR DESCRIPTION
**Pull request by dmichaels-harvard**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -29,11 +29,11 @@ classifiers = [
...
 [tool.poetry.dependencies]
-python = ">=3.7,<3.9"
-tibanna = "^3.3.1"
-dcicutils = "^7.0.0"
-boto3 = "^1.9.0"
-botocore = "^1.12.1"
+python = ">=3.7,<3.12"
+tibanna = "4.0.0.1b1"
+dcicutils = "7.12.0.1b1"
+boto3 = "^1.28.51"
```
Comment (netsettler) [inline pyproject.toml:35]: A good idea to include the corresponding `boto3-stubs` and `botocore-stubs` in dev-dependencies. I think those are supposed to have hinting or some other kind of debugging info.

**Review by netsettler — APPROVED** [review]
Specifying a higher poetry version seems important. Otherwise this is fine. My other comments are not blocking.

File: pyproject.toml (hunk 1 above)
Comment (dmichaels-harvard) [reply pyproject.toml]: Done

File: Makefile [hunk 2]
```diff
@@ -8,6 +8,9 @@ configure:
 lint:
 	flake8 tibanna_cgap
 
+build: 
+	poetry install
```
[the remark this answers is not in the document]
Comment (dmichaels-harvard) [reply Makefile:12]: Done

**Comment by alexander-veit** [comment]
There are more lambdas in `tibanna_smaht` and `tibanna_cgap` that need patching.

**Comment by alexander-veit** [comment]
We need to release Tibanna 5.0.0 first and then use that version here.

**Comment by alexander-veit** [comment]
The build in master is going to fail. As Tibanna 5.0.0 is not released. Also, this needed more testing on wolf.

**Comment by alexander-veit** [comment]
Oh, you released Tibanna? Did you upload the Docker image as well?

**Pull request merged**